src/mscorlib/shared/System/IO/Path.cs

```diff
@@ -126,23 +134,33 @@ public static string GetFileName(string path)
             if (path == null)
                 return null;
 
+            return new string(GetFileName(path.AsReadOnlySpan()));
```

Comment:
As @jkotas pointed out, this could allocate where it might not before. Return `path` if the returned span is the same length.